kpet/data.py

```diff
@@ -253,8 +279,15 @@ def __init__(self, data):
             ),
             data
         )
+        if self.sets is None:
+            self.sets = set()
+        elif isinstance(self.sets, list):
+            self.sets = set(self.sets)
         if self.pattern is None:
             self.pattern = Pattern({})
+        else:
+            self.pattern, sets = convert_sets(self.pattern)
+            self.sets |= sets
```

Comment:
If not for the comment above about pattern conversion, I'd ask you to move this out of the `else` branch and make it execute always. This would have made for simpler, easier to read code. Same in the `Suite`.